packages/framework/src/Model/Product/Search/FilterQueryFactory.php

```diff
@@ -14,4 +66,152 @@ public function create(string $indexName): FilterQuery
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    public function createListableProductsForBrandFilterQuery(
+        ProductFilterData $productFilterData,
+        string $orderingModeId,
+        int $page,
+        int $limit,
+        int $brandId
+    ): FilterQuery {
+        return $this->createFilterQueryWithProductFilterData($productFilterData, $orderingModeId, $page, $limit)
+            ->filterByBrands([$brandId]);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
+     * @param string $orderingModeId
+     * @param int $page
+     * @param int $limit
+     * @param string|null $searchText
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    public function createListableProductsForSearchTextFilterQuery(
+        ProductFilterData $productFilterData,
+        string $orderingModeId,
+        int $page,
+        int $limit,
+        ?string $searchText
+    ): FilterQuery {
+        $searchText = $searchText ?? '';
+
+        return $this->createFilterQueryWithProductFilterData($productFilterData, $orderingModeId, $page, $limit)
+            ->search($searchText);
+    }
+
+    /**
+     * @return string
+     */
+    public function getIndexName(): string
+    {
+        return $this->indexDefinitionLoader->getIndexDefinition(
+            ProductIndex::getName(),
+            $this->domain->getId()
+        )->getIndexAlias();
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    public function createSellableAndVisibleFilterQuery(): FilterQuery
+    {
+        return $this->filterQueryFactory->create($this->getIndexName())
+            ->filterOnlySellable()
+            ->filterOnlyVisible($this->currentCustomerUser->getPricingGroup());
+    }
+
+    /**
+     * @param int $categoryId
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    public function createProductFilterCountFilterQuery(int $categoryId, ProductFilterData $productFilterData): FilterQuery
+    {
+        $filterQuery = $this->createSellableAndVisibleFilterQuery();
+        $filterQuery->filterByCategory([$categoryId]);
+        $filterQuery = $this->addPricesAndStockFromFilterDataToQuery($productFilterData, $filterQuery);
+
+        return $filterQuery;
+    }
+
+    /**
+     * @param string|null $searchText
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    public function createProductSearchCountFilterQuery(?string $searchText, ProductFilterData $productFilterData): FilterQuery
+    {
+        $filterQuery = $this->createSellableAndVisibleFilterQuery();
+        $filterQuery->search($searchText);
```

Comment:
`$searchText` in this method is nullable, but `FilterQuery::search()` accept only string
not sure if it's not an error in the current version too

Reply:
fixed in 424a7bcc89ea946c848fe33b7703b96eda933c13